### ISSUE TITLE
creodias - improve Sentinel-1 default 'timeliness'

### ISSUE THREAD

**Issue by jdries** [Open-EO/openeo-geopyspark-driver#229]
On Creo, we're now have to do this:
s1properties["timeliness"] = lambda p: p == "NRT-3h|Fast-24h"

The problem is that the meaning of timeliness changed on a given date.

One way to resolve this is to do more advanced overlap handling in:
org.openeo.geotrellis.file.FileRDDFactory
The overlap handling can be reused from org.openeo.geotrellis.layers.FileLayerProvider#readKeysToRasterSources
To achieve this, a separate 'OverlapHandler' class can be created, so that we can even configure it per collection eventually.

**Comment by EmileSonneveld** [comment]
In `testCreaSentinal1` the timeliness filter removes 7 duplicates, while the properties-based deduping only removes 2 duplicates. (With properties-based, I mean the method based on startDate, publishedDate, orbitNumber, gemetry, ...) So before 2021-02-23 some more Features will be returned. Other test cases seem unaffected on first sight.
The timeliness filter is quite hacky and thus would be nice to remove. 
Passing `"NRT-3h|Fast-24h"` or nothing both give the same results.
-> Thus, [this code](https://github.com/Open-EO/openeo-opensearch-client/blob/9b497cdf1b90e933968245c194056535b1bd7e89/src/main/scala/org/openeo/opensearch/backends/CreodiasClient.scala#L71) can safely be removed. 
@jdries, does this reasoning sounds good to you? 

I implemented the dedup logic inside `openSearch.getProducts()`. It is a common place between normal usage and auto tests. I did not see any usages of FileRDDFactory.
